src/tools/moveedge.cpp

```diff
@@ -1,32 +1,76 @@
 #include "moveedge.h"
 #include "editorview.h"
 #include "edgeitem.h"
+#include "gripitem.h"
 #include <QMouseEvent>
 
 MoveEdgeTool::MoveEdgeTool()
 : BaseTool()
 {
 }
 
+void MoveEdgeTool::activated(EditorView* editor)
+{
+  QList<GripItem*> grips = editor->itemsOfType<GripItem>();
+  for (GripItem* grip : grips) {
+    grip->visible = false;
+  }
+}
+
+void MoveEdgeTool::deactivated(EditorView* editor)
+{
+  QList<GripItem*> grips = editor->itemsOfType<GripItem>();
+  for (GripItem* grip : grips) {
+    grip->visible = true;
+  }
+  for (EdgeItem* item : editor->itemsOfType<EdgeItem>()) {
+    item->hoverLeave();
+  }
+}
+
 bool MoveEdgeTool::mousePressEvent(EditorView* editor, QMouseEvent* event)
 {
+  pressed = true;
+  for (GripItem* grip : editor->selectedItems<GripItem>()) {
```

Comment:
Maybe instead of doing two loops, just deselect _everything_ using `editor->scene()->selectedItems()`?